pkgs/development/compilers/dotnet/sign-apphost.nix

```diff
@@ -0,0 +1,10 @@
+{ substituteAll
+, callPackage
+}:
+let
+  sigtool = callPackage ./sigtool.nix {};
+
+in substituteAll {
+  src = ./sign-apphost.proj;
+  codesign = "${sigtool}/bin/codesign";
```

Comment:
I'll take a look.  My instinct is that it would be better to write a wrapper for compatibility, like the one included with `sigtool`.  We currently just replace `/usr/bin/codesign`.  Without the wrapper we'd have to patch this code:

```
            if (_macosCodesign && RuntimeInformation.IsOSPlatform(OSPlatform.OSX) && HostModelUtils.IsCodesignAvailable())
            {
                var (exitCode, stdErr) = HostModelUtils.RunCodesign("-s -", bundlePath);
                if (exitCode != 0)
                {
                    throw new InvalidOperationException($"Failed to codesign '{bundlePath}': {stdErr}");
                }
            }

            return bundlePath;

            // Remove mac code signature if applied before bundling
            static void RemoveCodesignIfNecessary(string bundlePath)
            {
                Debug.Assert(RuntimeInformation.IsOSPlatform(OSPlatform.OSX));
                Debug.Assert(HostModelUtils.IsCodesignAvailable());

                // `codesign -v` returns 0 if app is signed
                if (HostModelUtils.RunCodesign("-v", bundlePath).ExitCode == 0)
                {
                    var (exitCode, stdErr) = HostModelUtils.RunCodesign("--remove-signature", bundlePath);
                    if (exitCode != 0)
                    {
                        throw new InvalidOperationException($"Removing codesign from '{bundlePath}' failed: {stdErr}");
                    }
                }
            }
```